File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/ClinicModule/MedicalHistory.cs
﻿using LMS_CMS_DAL.Models.Domains.LMS;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LMS_CMS_DAL.Models.Domains.ClinicModule
{
    public class MedicalHistory : AuditableEntity
    {
        [Key]
        public long Id { get; set; }

        //[Required(ErrorMessage = "School ID is required")]
        [ForeignKey("School")]
        public long? SchoolId { get; set; }
        public School? School { get; set; }

        //[Required(ErrorMessage = "Grade ID is required")]
        [ForeignKey("Grade")]
        public long? GradeId { get; set; }
        public Grade? Grade { get; set; }

        //[Required(ErrorMessage = "Classroom ID is required")]
        [ForeignKey("Classroom")]
        public long? ClassRoomID { get; set; }
        public Classroom? Classroom { get; set; }

        //[Required(ErrorMessage = "Student ID is required")]
        [ForeignKey("Student")]
        public long? StudentId { get; set; }
        public Student? Student { get; set; }

        public string? Details { get; set; }

        public string? FirstReport { get; set; }
        public string? SecReport { get; set; }

        public int? Attached { get; set; } 

        public string? PermanentDrug { get; set; }

        public DateTime? Date { get; set; }
    }
}

File: LMS_CMS_Frontend/src/app/Models/Clinic/MedicalHistory.ts
export class MedicalHistory {
  [key: string]: any;

  constructor(
    public id: number,
    public schoolId: number,
    public school: string = "",
    public gradeId: number,
    public grade: string = "",
    public classRoomID: number,
    public classRoom: string = "",
    public studentId: number,
    public student: string = "",
    public details: string,
    public permanentDrug: string,
    public insertedAt: string,
    public firstReport: File | null,
    public secReport: File | null,
    public updatedAt?: string,
    public doctorId?: number,
    public doctorName?: string
  ) {}
}